frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -111,12 +117,19 @@ export function isPipelineSpec(templateString: string) {
 export function getContainer(componentSpec: ComponentSpec, templateString: string) {
   const executionLabel = componentSpec?.executorLabel;
 
-  const jsonTemplate = getPipelineDefFromYaml(templateString);
-  const deploymentSpec = jsonTemplate['deploymentSpec'];
+  const pipelineSpecYAML = getPipelineDefFromYaml(templateString);
+  const pipelineSpec = PipelineSpec.fromJSON(pipelineSpecYAML);
+  const deploymentSpecObj = pipelineSpec.deploymentSpec;
```

Comment:
if `pipelineSpec` is empty, I think this line will throw error. Consider throwing exception first by checking `pipelineSpec`, or add a question mark like `const deploymentSpecObj = pipelineSpec?.deploymentSpec;`